tests/behat/behat_repository_ocis.php

```diff
@@ -95,38 +140,136 @@ private function get_password_for_user(string $username): string {
-        if (strtolower($space) === "personal") {
-            $this->create_file_in_personal_space($user, $file);
+    private function create_file_in_personal_space(string $user, string $file, string $body): void {
+        $client = $this->get_client($user);
+        $response = $client->request('PUT', "/dav/files/$user/$file", $body);
+        if (!in_array($response['statusCode'], [201, 204])) {
+            throw new Exception("Error creating resource in ocis " . var_dump($response['statusCode']));
         }
         $this->createdfiles[] = ["file" => $file, "user" => $user];
     }
 
-    private function get_client($user, $password): Client {
-        $settings = [
-            'baseUri' => getenv('MOODLE_OCIS_URL'),
-            'userName' => $user,
-            'password' => $password,
-        ];
-        return new Client($settings);
-    }
-
-
-    private function create_file_in_personal_space($user, $file) {
-        $client = $this->get_client($this->get_actual_username($user), $this->get_password_for_user($user));
-        $response = $client->request('PUT', "/dav/files/$user/$file");
-        if (!in_array($response['statusCode'], [201, 204])) {
-            throw new Exception("Error creating resource in ocis " . var_dump($response['statusCode']));
+    /**
+     * @param $spaceid
+     *
+     * @return void
+     * @throws Exception
+     */
+    public function delete_space($spaceid): void {
+        $client = $this->get_admin_client();
+        $header = ["Purge" => "T"];
+        $disabledrive = $client->request('DELETE', "/graph/v1.0/drives/$spaceid");
+        $deletedrive = $client->request('DELETE', "/graph/v1.0/drives/$spaceid", null, $header);
+        if ($disabledrive['statusCode'] && $deletedrive['statusCode'] !== 204) {
+            throw new Exception('Error deleting drive');
         }
     }
 
     private function delete_file_in_personal_space($user, $file) {
-        $client = $this->get_client($this->get_actual_username($user), $this->get_password_for_user($user));
+        $client = $this->get_client($user);
         $response = $client->request('DELETE', "/dav/files/$user/$file");
         if ($response['statusCode'] !== 204) {
             throw new Exception("Error deleting resource in ocis " . var_dump($response['statusCode']));
         }
     }
+
+    /**
+     * @param string $space
+     * @param string $resource
+     * @param string $body
+     *
+     * @return void
+     * @throws Exception
+     */
+    public function create_resource_in_project_space(string $space, string $resource, string $body): void {
+        $client = $this->get_admin_client();
+        $spaceid = $this->get_space_id($space);
+        $response = $client->request('PUT', "dav/spaces/$spaceid/$resource", $body);
+        if ($response['statusCode'] != 201) {
+            throw new Exception("Error creating resource in ocis " . var_dump($response['statusCode']));
```

Comment:
```suggestion
            throw new Exception("Error creating resource in ocis project space '$space', status was " . var_dump($response['statusCode']));
```

will it help to give more information in the exception message?